test/media/drm_engine_unit.js

```diff
@@ -677,6 +677,35 @@ describe('DrmEngine', () => {
           .toBeRejectedWith(expected);
     });
 
+    it('uses key system IDs from keySystemsMapping config', async () => {
+      manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        manifest.addVariant(0, (variant) => {
+          variant.addVideo(1, (stream) => {
+            stream.encrypted = true;
+            stream.addDrmInfo('drm.abc');
+          });
+        });
+      });
+
+      setDecodingInfoSpy([]);
+
+      config.keySystemsMapping['drm.abc'] = 'drm.def';
+      drmEngine.configure(config);
+
+      const variants = manifest.variants;
+      await expectAsync(
+          drmEngine.initForPlayback(variants, manifest.offlineSessionIds))
+          .toBeRejected();
+
+      expect(drmEngine.initialized()).toBe(false);
```

Comment:
FMI, this scenario is set to fail DRM initialization because in L690, it was set to fail all key systems? 

But even then, it still reaches these lines: https://github.com/shaka-project/shaka-player/blob/c935cc17703297a44b3ce3bda75d8f2ea37f4147/lib/media/drm_engine.js#L351-L361 which is why L701-705 show the decodingInfoSpy is called with the mapping information?